ocfweb/docs/views/servers.py

```diff
@@ -61,6 +61,18 @@ def english_type(self):
     def has_munin(self):
         return self.type in ('hypervisor', 'vm', 'server', 'desktop')
 
+    def __key(self):
+        if self.type == 'hypervisor':
+            return 'a' + self.hostname
+        if self.type == 'server':
+            return 'b' + self.hostname
+        if self.type == 'desktop':
+            return 'z' + self.hostname
+        return 'c' + self.type + self.hostname
```

Comment:
I'd use a tuple here, so instead of `'a' + hostname`, it would be `(1, hostname)`. Makes it more clear that this is for sorting.

Also, I'd use if/elif/else here, so all of the return statements are at the same level of indentation.

Reply:
This feels really hacky. I would not use a tuple either. I think a class to abstract this away would make it more simple.

```python
from enum import Enum

class Machine(Enum):
    HYPERVISOR = 1 
    SERVER = 2 
    OTHER = 3 
    DESKTOP = 4 

    @classmethod
    def from_string(cls, machine):
        if machine == 'hypervisor':
            return Machine.HYPERVISOR
        elif machine == 'server':
            return Machine.SERVER
        elif machine == 'desktop':
            return Machine.DESKTOP
        else:
            return Machine.OTHER

    def __lt__(self, other):
        return self.__class__ is other.__class__ and self.value < other.value
```

Now you can just do `Machine.from_string(self.type) < Machine.from_string(other_host.type)` in `__lt__`.

Reply:
@keur what you propose could be done more succinctly with a dictionary mapping type strings to numbers. Additionally, you would have to teach `__lt__` to compare type strings and hostnames as well as the enum value. Which makes the final solution about equal in complexity.

Reply:
After coding up different approaches I found this one being possibly the easiest to understand. It's a combination of using a dictionary to store the order and using tuples as keys.
https://github.com/ocf/ocfweb/pull/406/commits/c83f8c81b703657e01f800cacd84cc6f80d2451a